scanning/disa-alignment/shared.py

```diff
@@ -52,11 +52,11 @@ def comparison_report(comparison_output):
       Same result format:      CCE CCI - DISA_RULE_ID SSG_RULE_ID     RESULT
       Different result format: CCE CCI - DISA_RULE_ID SSG_RULE_ID     SSG_RESULT - DISA_RESULT
     """
-    result_regex = re.compile(r'[\w-]+ [\w-]+ - [\w-]+ (\w*)\s+(\w+)(?: - *(\w+))*')
+    result_regex = re.compile(r'([\w-]+) [\w-]+ - [\w-]+ (\w*)\s+(\w+)(?: - *(\w+))*')
     for match in result_regex.finditer(comparison_output):
-        rule_id, ssg_result, disa_result = match.groups()
+        rule_cce, rule_id, ssg_result, disa_result = match.groups()
         if not rule_id:
-            rule_id = 'rule_id_not_found'
+            rule_id = f'ID not found in ARF ({rule_cce})'
```

Comment:
> Are you sure you want this as a result test name (and not a note)?

Good point, it's not ideal for waiving. I can change that for CCE
```
/scanning/disa-alignment/oscap/CCE-12345-6
```
^ still not ideal, but we don't have better option at the moment. Test parses output of script that compares ARFs and the script doesn't print rule id if it doesn't find it.

> It kinds of made me realize I don't really know how to interpret the output of this test - I thought it was a diff, but in that case, it would make sense to do something like
PASS    /scanning/disa-alignment/oscap/some_passing_rule_in_both
FAIL    /scanning/disa-alignment/oscap/some_inconsistent_rule_in_both    (SSG: notapplicable, DISA: pass)
FAIL    /scanning/disa-alignment/oscap/some_rule_only_in_one    (SSG: pass, DISA: missing, CCE-12345-6)

1st example is met.
For the 2nd example, well, RHEL8 DISA's benchmarks don't have CPE capability (written in comment) as we do and so DISA reports `pass` (e.g. when package related to service is not installed). But in recently released RHEL9 DISA benchmark, they do applicability check but only for few rules. So I wonder if removing the `elif ssg_result == 'notapplicable' and disa_result == 'pass':` condition and moving rules to waivers makes sense.
3rd example, that would be ideal. The problem here is we don't have the `some_rule_only_in_one`.